### PR TITLE
[TASK] PWA area, Enable / Disable PWA, Dynamically set SERVER variables, Default frontend fallback

### PR DESCRIPTION
**Pull request by dheesbeen**
- Added PWA (frontend) area
- Added setting to disable or enable storeview per store
- Added setting to skip certain routers and load from default magento frontend, 
- Dynamically set _SERVER variables for pwa application

### CODE REVIEW

**Comment by dheesbeen** [comment]
@tjwiebell I've created a pull request for the features mentioned in issue #11

**Comment by dheesbeen** [comment]
@bbatsche @tjwiebell 
Have you had the opportunity to look at this? If the adjustments are not relevant for this repo let me know. Then I will continue on a different version. Currently I also implement the transfer of the state (session) between the PWA and the regular Magento frontend.  Let me know! Thanks a lot. Derrick

**Comment by zetlen** [comment]
Hi @dheesbeen . Thanks so much for volunteering this.

I should start by saying I'm not a PHP developer, so I can't have the final word on code review for this. But I can ask a few high-level questions.

1. **Why would you enable certain controllers, rather than certain route patterns**?
2. **Can you describe what uses cases which need the store view selector?** We want the connection between a deployed PWA and the Magento store to be as _one-way_ as possible. The Magento instance functioning "headlessly" should have minimal knowledge of the frontend. This is to avoid tight coupling and also to allow for many-to-one scenarios, where multiple PWAs can use the same backing Magento instance with no problems. (The key use case for this is a frontend developer team working on a new project, who can all use a shared internal staging store.)
3. **Can you explain why we would need the `_SERVER` variables for the PWA application?** Same as above...the PWA should not have to use the same request/response object as the rest of the Magento application!

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -11,13 +11,23 @@
...
                 </field>
+                <field id="enabled" showInDefault="1" showInStore="1" showInWebsite="1" sortOrder="10" translate="label,comment" type="select">
+                    <label>Enabled</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <comment>Enabled PWA frontend for this store</comment>
+                </field>
+                <field id="front_names_to_skip" showInDefault="1" showInStore="1" showInWebsite="1" sortOrder="30" translate="label,comment" type="textarea">
+                    <label>Front Names to skip</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <comment>Separated bij newline. You can load certain url's from the default magento frontend.</comment>
```
Comment (paliarush) [inline etc/adminhtml/system.xml]:
```suggestion
                    <comment>Separated by newline. You can load certain url's from the default magento frontend.</comment>
```

**Comment by zetlen** [comment]
@tjwiebell Has most of this PR been covered in #19 and this can be closed, or is there other stuff to excavate?

**Comment by tjwiebell** [comment]
@zetlen There is some logic remaining about automatic environment variable setting, but this is what gave us pause about merging this to begin with. I propose closing and asking @dheesbeen to open a new PR with that functionality, and justification for their use cases, and it can be reviewed during our routine grooming ceremony.

**Pull request closed without merging**